### PR TITLE
Improve release process

### PR DESCRIPTION
**Pull request by dliappis**
Commit `Dockerfile`, `docker-compose` and new `release-in-docker` make target
to allow running the release process from within a Docker container.

The container tries to use the same username/uid as the local user and
bind-mounts the needed files for gpg, ssh-agent and github release
token.

The execution of `make it` is still a task left outside of the container
primarily due to execution time on macOS.

Relates #483

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.6
+
+ARG NEW_USER_UID
+ARG NEW_USER
+
+RUN apt-get -y update && \
+    apt-get install -y vim openssh-client python-tox && \
```
Comment (danielmitterdorfer) [inline Dockerfile]: out of curiosity: Why do we need `vim`?

Comment (dliappis) [reply Dockerfile]: This was just in case something goes weird inside the container and we need to edit a file, which `exec`'ed inside the container. Since we are binding everything locally this is generally not needed, but I though maybe there's this edge case where we need to quickly editing something (had to do it myself :) ).

**Comment by dliappis** [comment]
Thanks for the review; yes I am afraid we can only properly test this on a real release; there are a few things though to try esp. on a macOS system before hand esp. if the the ssh keys get passed correctly through the agent and gpg.

+1: for rebase here.

File: Dockerfile (hunk 1 above)
Comment (danielmitterdorfer) [reply Dockerfile]: I see. Makes sense.

**Comment by dliappis** [comment]
Closing this, as passing ssh keys inside a container when keys are cached on macOS using keychain is not trivial. May revisit this in the future.
